HTML/EN/settings/server/behavior.html

```diff
@@ -10,28 +13,64 @@
 		[% END %]
 	[% END %]
 
-	[% IF prefs.pref_useUnifiedArtistsList; WRAPPER settingSection %]
-		[% WRAPPER settingGroup title="SETUP_COMPOSERINARTISTS" desc="SETUP_COMPOSERINARTISTS_DESC" %]
-			<input type="checkbox" [% IF prefs.pref_trackartistInArtists %]checked="1" [% END %] class="stdedit" name="pref_trackartistInArtists" id="trackartistInArtists" value="1" />
-			<label for="trackartistInArtists" class="stdlabel">[% "TRACKARTIST" | string %]</label><br>
-
-			<input type="checkbox" [% IF prefs.pref_composerInArtists %]checked="1" [% END %] class="stdedit" name="pref_composerInArtists" id="composerInArtists" value="1" />
-			<label for="composerInArtists" class="stdlabel">[% "COMPOSER" | string %]</label><br>
-
-			<input type="checkbox" [% IF prefs.pref_conductorInArtists %]checked="1" [% END %] class="stdedit" name="pref_conductorInArtists" id="conductorInArtists" value="1" />
-			<label for="conductorInArtists" class="stdlabel">[% "CONDUCTOR" | string %]</label><br>
+	[% WRAPPER settingSection %]
+		[% WRAPPER settingGroup title="SETUP_ROLESINARTISTS" desc="SETUP_ROLESINARTISTS_DESC" %]
+		<table>
+			<tr>
+			[% SET counter = 0 %]
+			[% FOREACH role = menuRoles %]
+				[% IF !role.hidden %]
+					[% SET counter = counter + 1 %]
+					<td><input type="checkbox" [% IF role.selected %]checked="1" [% END %] class="stdedit" name="pref_[% role.name %]InArtists" id="[% role.name %]InArtists" value="1" />
+					<label for="[% role.name %]InArtists" class="stdlabel">[% role.name | string %]</label></td>
+					<td />
+					[% IF counter % numberOfRoleCols == 0 %] </tr><tr> [% END %]
+				[% ELSE %]
+					<input type="hidden" name="pref_[% role.name %]InArtists" value="[% role.selected | html %]" />
+				[% END %]
+			[% END %]
+			</tr>
+		</table>
+		[% END %]
+	[% END %]
```

Comment:
Shouldn't this block be conditional? Right now I'd only get the checked items in "two list" mode. And then, if I deselect them, they would stay? I don't get the logic...

Reply:
Previously that block only applied for the combined list, to show the options TRACKARTIST/COMPOSER/CONDUCTOR/BAND. Now it will be used in both modes, so that user-defined roles can be shown. But the `role.hidden` flag will suppress the 4 standard roles when you're using two separate lists, to retain current behaviour.

Reply:
Good point! I could just condition it so the section doesn't appear at all in using 2 lists with no user-defined roles.